app/controllers/signup_controller.rb

```diff
@@ -71,6 +69,18 @@ def verify_by_token
                 end)
   end
 
+  def trusted_student
+    handle_with(SignupTrustedStudent,
+                signup_state: signup_state,
+                session: self,
+                success: lambda do
+                  redirect_to action: :profile
+                end,
+                failure: lambda do
+                  redirect_to action: :start
+                end)
+  end
```

Comment:
Instead of a dedicated `trusted_student` action and handler, can we just use the existing flow and look to the signup state to know whether we can skip email and password solicitation?

Reply:
no, because the regular flow creates the user account after a password is added (in https://github.com/openstax/accounts/blob/master/lib/omniauth/strategies/custom_identity.rb#L131-L134)

Since student's don't get a password that's never called.  I extracted the "create user from signup" into a routine and am calling it from both handlers.

Reply:
I've changed my mind on this, and have extracted some of the logic into a `SignupTrustedStudent` routine that I'm calling from the `SignupStart` handler.  That seems to work well, but I'm not super happy with how large `SignupStart` is now 😁